tools/rust_api/src/ffi.rs

```diff
@@ -340,6 +376,12 @@ pub(crate) mod ffi {
         fn create_value_int128_t(high: i64, low: u64) -> UniquePtr<Value>;
         fn create_value_uuid_t(high: i64, low: u64) -> UniquePtr<Value>;
         fn create_value_internal_id(offset: u64, table: u64) -> UniquePtr<Value>;
+        fn create_value_decimal(
```

Comment:
I suggest separating decimal creation into two parts
One part could be a user facing function that takes `value: i128, scale: u32` and determines the high, low, precision, and scale, to call the less user friendly `create_value_decimal`, or some variant.
Internally, a DECIMAL(10, 3) can easily be cast to a DECIMAL(18, 3) so you can resolve the precision to be the minimal precision that fits.

Reply:
From the rust side, you create decimals using the [`Value::Decimal { value: rust_decimal::Decimal, precision: u32 }`](https://github.com/kuzudb/kuzu/pull/3827/files#diff-fba2944519c68ae7315eeeb0cbbeb726e56030ab9f774d5eeb9b8d90502a716fR273-R276) enum. [E.g.](https://github.com/kuzudb/kuzu/pull/3827/files#diff-fba2944519c68ae7315eeeb0cbbeb726e56030ab9f774d5eeb9b8d90502a716fR1276) (note that the `dec!` macro is part of a separate crate, rust_decimal_macros).
The `ffi::create_value_decimal` is just used as part of the internal conversion to a C++ Value which is used when Rust Values are passed to `Connection::execute`.
I'd added the precision explicitly as I was assuming that when passed to `Connection::execute`, the precision needs to match the parameter type. Or will it be cast automatically? If so we could change it to `Value::Decimal(rust_decimal::Decimal)`.

Reply:
Ah I see. I'm fairly certain you don't need to match the precision explicitly. If you're talking about something where a column or function expects a decimal with a higher precision, but as a parameter, you pass a decimal with a lower precision, then there would be an implicit cast.

Reply:
I figured that out through some experimentation. I've since updated the PR to remove the explicit precision.